waltz-ng/client/data-flow/components/data-flows-tabgroup/data-flows-tabgroup.js

```diff
@@ -139,7 +140,78 @@ function prepareGraphTweakers(dataFlowUtilityService,
 }
 
 
-function controller($scope, dataFlowUtilityService) {
+function mkEntityNameCell(entityDataObjectField, valueField, columnHeading, entityNavViewName) {
+    return {
+        field: valueField,
+        displayName: columnHeading,
+        cellTemplate: `<div class="ui-grid-cell-contents">\n<a ui-sref="main.${entityNavViewName}.view ({ id: row.entity[\'${entityDataObjectField}\'][\'id\']})" ng-bind="COL_FIELD">\n</a>\n</div>`
+    };
+}
+
+
+function groupDecoratorsByFlowId(decorators = [], displayNameService) {
+    return _.chain(decorators)
+        .filter(dc => dc.decoratorEntity.kind === 'DATA_TYPE')
+        .map(dc => _.assign(
+                    {},
+                    {
+                        dataFlowId: dc.dataFlowId,
+                        dataType: {
+                            id: dc.decoratorEntity.id,
+                            name: displayNameService.lookup('dataType', dc.decoratorEntity.id)
+                        },
+                        authSourceRating: dc.rating
+                    }))
+        .groupBy('dataFlowId')
+        .value();
+}
+
+
+function prepareGridData(flows = [], decorators = [], displayNameService) {
+    const groupedDecorators = groupDecoratorsByFlowId(decorators, displayNameService);
+    return _.flatMap(flows,
+                    f => _.map(groupedDecorators[f.id],
+                                dc => _.assign(
+                                        {},
+                                        f,
+                                        { dataType: dc.dataType },
```

Comment:
why do this as three additional objects, why not compress final 3 into `{ dataType: ... , authSourceRating:...., rowStyle: ... }` ?